R/counts_per_million.R

```diff
@@ -0,0 +1,49 @@
+#' Generating counts per million from SingleCellExperiment object
+#'
+#' @param input_sce_obj A SingleCellExperiment object read from RDS
+#' @param output_dir A character vector of counts per million path
+#' @param hd5_file_dir A character vector of HDF5 file path after converting from RDS
+#' @return A directory stores counts per million 
+#' @importFrom SingleCellExperiment SingleCellExperiment
+#' @importFrom dplyr tbl
+#' @importFrom glue glue
+#' @importFrom HDF5Array saveHDF5SummarizedExperiment
+#' @importFrom purrr map
+#' @noRd
+get_counts_per_million <- function(input_sce_obj, output_dir, hd5_file_dir) {
+  
+  # Create directories
+  output_dir |>  dirname() |> dir.create( showWarnings = FALSE, recursive = TRUE)
+
+  # Save SCE to the cache directory counts folder
+  input_sce_obj |> saveHDF5SummarizedExperiment(hd5_file_dir)
+  
+  data <- input_sce_obj
+  
+  # Avoid completely empty cells
+  col_sums <- colSums(as.matrix(data@assays@data$X))
```

Comment:
Do you need to use `as.matrix` here? It seems likely that `colSums` will work directly, but it will be much more efficient because it won't load the entire matrix from disk.

Also as I have noted elsewhere, you should avoid using `@`.